File: 1200-Minimum-Abs-Difference/1200.java
class Solution {
  public List<List<Integer>> minimumAbsDifference(int[] arr) {
    Arrays.sort(arr);
    List<List<Integer>> res = new LinkedList<>();
    int min = 99999999;
    for (int i = 1; i < arr.length; i++) {
      int tmp = arr[i] - arr[i - 1];
      if (tmp < min) {
        min = tmp
        res.clear()
      }
      if (tmp == min) {
        List<Integer> item = new ArrayList<>(2);
        item.add(arr[i - 1]);
        item.add(arr[i]);
        res.add(item);
      }
    }
    return res;
  }
}

File: 1101-1200/1200-Minimum-Abs-Difference/1200.py
class Solution:
  def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
    arr.sort()
    Hash = {}
    for i in range(len(arr) - 1):
      distance = arr[i + 1] - arr[i]
      if Hash.get(distance) is None:
        Hash[distance] = [[arr[i], arr[i+1]]]
      else:
        Hash[distance].append([arr[i], arr[i+1]])
    return Hash[min(Hash.keys())]